corehq/warehouse/transforms/sql/location_dim.sql

```diff
@@ -0,0 +1,130 @@
+INSERT INTO {{ location_dim }} (
+    domain,
+
+    location_id,
+    sql_location_id,
+
+    level,
+    location_level_0,
+    location_level_1,
+    location_level_2,
+    location_level_3,
+    location_level_4,
+    location_level_5,
+    location_level_6,
+    location_level_7,
+
+    name,
+    site_code,
+    external_id,
+    supply_point_id,
+    is_archived,
+    latitude,
+    longitude,
+
+    location_type_name,
+    location_type_code,
+    location_type_id,
+
+    location_last_modified,
+    location_created_on,
+
+    dim_last_modified,
+    dim_created_on,
+    -- TODO: Figure out how to handle deletes when we actually hard deleted the model
+    deleted
+)
+WITH RECURSIVE location_hierarchy(
```

Comment:
curious if you think it might be easier to first insert/update the dim with all the single location's info, and then update all the foreign keys later. i think we could use one update query to refresh all the foreign key ids using an update...from query since the table should be small enough.